pkg/multicluster/register/registrant.go

```diff
@@ -289,7 +301,7 @@ func (c *clusterRegistrant) RegisterClusterWithToken(
 	token string,
 	opts Options,
 ) error {
-	return c.RegisterProviderClusterWithToken(ctx, masterClusterCfg, remoteClientCfg, token, opts, nil)
+	return c.RegisterProviderClusterWithToken(ctx, masterClusterCfg, remoteClientCfg, token, opts, nil, nil, nil)
```

Comment:
what's with the nils? just curious. it would help to use a struct as a param here, so the strut fields could be named and commented with a bit of insight into what's optional